javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/SafeServiceLoader.java

```diff
@@ -54,4 +53,43 @@ public static <T extends Ordered> List<T> loadOrdered(Class<T> serviceClass) {
   }
 
   private SafeServiceLoader() {}
+
+  private static class SafeIterator<T> implements Iterator<T> {
+    private final Iterator<T> delegate;
+
+    SafeIterator(Iterator<T> iterator) {
+      delegate = iterator;
+    }
+
+    private static void handleUnsupportedClassVersionError(
+        UnsupportedClassVersionError unsupportedClassVersionError) {
+      logger.log(
+          FINE,
+          "Unable to load instrumentation class: {0}",
+          unsupportedClassVersionError.getMessage());
+    }
+
+    @Override
+    public boolean hasNext() {
+      // jdk9 and newer throw UnsupportedClassVersionError in hasNext()
```

Comment:
oh interesting, so we would have run into this problem on Java 11 if we introduced Java 17-only instrumentation

do you know if the underlying iterator will move on item hasNext when it throws exception, so that we will continue iterating over it from that point?

Reply:
I tested it manually and believe that `hasNext` will move to the next element on exception. Relevant code is in
https://github.com/openjdk/jdk11u/blob/b928a88fd564fab88f00086a62c0695ed5cc9353/src/java.base/share/classes/java/util/ServiceLoader.java#L1181 `pending` iterator is advanced before `Class.forName`